File: Swift-C++-IR-Comparison/test/closures/Swift/File.swift
// As an optimization, Swift may instead capture and store a copy of
//  a value if that value isn’t mutated by a closure, and if the value
//  isn’t mutated after the closure is created.

// Swift also handles all memory management involved in disposing of
// variables when they’re no longer needed.

@main
struct MyMain {

    static func main() {
        var incrementer: (Int) -> Int = (x:Int) { return x + 5 }
        func explicitClosure(myFunction: (Int) -> Int) {
            myFunction(5)
        }
        explicitClosure(myFunction: incrementer)
        explicitClosure {
            return x + 5
        }
    }
}

File: AutoTranspiledTests/Swift-to-Cpp/try-1/File.cpp

# include <iostream>

// As an optimization, C++ may instead capture and store a copy of
//  a value if that value isn’t mutated by a closure, and if the value
//  isn’t mutated after the closure is created.

// C++ also handles all memory management involved in disposing of
// variables when they’re no longer needed.

int main(){
    auto incrementer = [](int x){ return x + 5; };
    void explicitClosure([](int)(int)myFunction) {
        myFunction(5);
    }
    explicitClosure(incrementer);
    explicitClosure([](int x){ return x + 5; };);
    return 0;
}